### ISSUE TITLE
Objective.evaluate() fails if the objective function is not defined.

### ISSUE THREAD

**Issue by snowberryfield** [snowberryfield/cpp_metaheuristics#41]
The `Objective.evaluate()` method fails if the objective function is not defined because the `FixedSizedHashMap` object would not be built for that case. The step `m_expression.setup_fixed_sensitivities();` in the following code in `objective.h` should be moved into `Model.setup_fixed_sensitivities()` method.

```c++
    /*************************************************************************/
    inline constexpr void setup(
        const Expression<T_Variable, T_Expression> &a_EXPRESSION) {
        this->initialize();
        m_is_linear  = true;
        m_expression = a_EXPRESSION;
        m_expression.setup_fixed_sensitivities();
    }
```

**Comment by snowberryfield** [comment]
An alternative idea to fix the bug is to add conditional guards for the methods `m_objective.evaluate()` and `m_objective.update()` in `model.h` using the existing flag `m_is_defined_objective`.

**Comment by snowberryfield** [comment]
This bug was fixed by 5c2b73d.
